pybossa/view/bloomberg.py

```diff
@@ -104,3 +113,16 @@ def handle_bloomberg_response():
         current_app.logger.exception('BSSO login error')
         flash(gettext('We were unable authenticate and log you into an account. Please contact a Gigwork administrator.'), 'error')
         return redirect(url_for('home.home'))
+
+
+def get_user_data_access_level(user_attributes):
+    """Reads firm id to user type mappings from settings_upref_mdata and
+    returns the access type"""
+    firm_num_to_type = current_app.config.get('FIRM_TO_TYPE')
+    firm_num = user_attributes.get('firmId', [None])[0]
+    if current_app.config.get('PRIVATE_INSTANCE') and int(firm_num):
+        return ['L2'] if int(firm_num) in firm_num_to_type.keys() else ['L4']
+    else:
+        admin_msg = generate_bsso_account_warning(user=user_attributes, admins_emails=current_app.config['ADMINS'], access_type="BSSO")
+        mail_queue.enqueue(send_mail, admin_msg)
```

Comment:
function get_user_data_access_level should only return data access level as its intended. Another reason is that it could result into false notification as account creation happens later that could fail for some reason. Hence, warning notification to be sent after [create account](https://github.com/bloomberg/pybossa/pull/517/files#diff-ca230f868b3bb408c2217967bf8b5452R102) checking data access level type